### PR TITLE
Feature/upload azure

### PR DESCRIPTION
**Pull request by bensilvan**

### CODE REVIEW

File: src/components/User/Mission/EditMission.js [hunk 1]
```diff
@@ -18,14 +20,51 @@ export default function EditMission(props) {
...
 
   const sendEditRequest = async () => {
+    const missionsFilesWithId = requestState.files.map(file => {
+      return {
+        browserFile: file,
+        id: `${ID()}_${file.name}`
+      }
+    })
+
+    var tasksFileWithID = {};
```
Comment (nachtiomer) [inline src/components/User/Mission/EditMission.js:30]: change to const and don't use var (ever)

File: src/components/User/Mission/NewMission.js [hunk 2]
```diff
@@ -1,27 +1,66 @@
...
+      if (task.files.length > 0) {
+        tasksFileWithID[task._id] = task.files.map(file => ({ browserFile: file, id: `${ID()}_${file.name}` }))
+      }
+    });
+
+    var allFiles = [...missionsFilesWithId];
+    for (var files of Object.values(tasksFileWithID)) {
+      allFiles = allFiles.concat([...files]);
+    }
+
```
Comment (nachtiomer) [inline src/components/User/Mission/NewMission.js:37]: tasks.files is an array so it won't hurt you to iterate all files like each task has files.
you can concat all files of mission and tasks.

File: src/components/User/Mission/NewMission.js [hunk 3]
```diff
@@ -1,27 +1,66 @@
...
+    const missionFiles = missionsFilesWithId.map(file => ({
+      _id: file.id,
+      name: file.browserFile.name,
+      type: file.browserFile.type,
+      size: file.browserFile.size
+    }))
```
Comment (nachtiomer) [inline src/components/User/Mission/NewMission.js:43]: create a function that recieve a file and returns his db object
there's code duplication

File: src/components/User/Mission/Request/TasksForm.js [hunk 4]
```diff
@@ -159,12 +162,14 @@ export default function TasksForm({submitTask, formError}) {
                   required
                   style={{width: "100%", marginBottom: "24px"}}
               />
+              <FilesUpload files={files} setFiles={setFiles} />
```
Comment (nachtiomer) [inline src/components/User/Mission/Request/TasksForm.js:165]: limit to 1 file

File: src/utils/api.js [hunk 5]
```diff
@@ -1,4 +1,6 @@
 const {REACT_APP_API} = process.env;
+const { BlobServiceClient } = require('@azure/storage-blob')
+const HOST = "https://masal.blob.core.windows.net";
```
Comment (nachtiomer) [inline src/utils/api.js:3]: very abstract name, be more specific and think if there's another place to declare this variable

File: src/utils/api.js [hunk 6]
```diff
@@ -139,3 +141,21 @@ export const fetchDashboards = async () => {
   const response = await fetch(`${REACT_APP_API}/stats/dashboards`);
   return await response.json();
 }
+
+export const uploadFiles = async (files) => {
+  const res = await fetch(`${REACT_APP_API}/azure/writeAccessDetails`)
```
Comment (nachtiomer) [inline src/utils/api.js:146]: add notes on this function to describe how it works

File: src/utils/api.js (hunk 5 above)
Comment (bensilvan) [reply src/utils/api.js:3]: this const is relevant only for this file and never be relevant to others.
I can put it in a configuration file but create one for one const is unnecessary

File: src/utils/api.js (hunk 6 above)
Comment (bensilvan) [reply src/utils/api.js:146]: the code explains what it does (fetch token and container, get container, write a blob.. )

File: src/components/User/Mission/NewMission.js [hunk 7]
```diff
@@ -1,27 +1,65 @@
...
+        tasksFileWithID[task._id] = task.files.map(file => ({ browserFile: file, id: `${ID()}_${file.name}` }))
+      }
+    });
+
+    let allFiles = [...missionsFilesWithId];
+    for (let files of Object.values(tasksFileWithID)) {
+      allFiles = allFiles.concat([...files]);
+    }
+
+    const fileToDbModel = (file) => ({
```
Comment (nachtiomer) [inline src/components/User/Mission/NewMission.js:38]: it should be outside (above) the component, it's not a part of it's flow

File: src/utils/api.js (hunk 6 above)
Comment (nachtiomer) [reply src/utils/api.js:146]: yeah but still you don't want each developer to start reading all about azure, add some comments that describes the flow and how azure works

File: src/components/FilesDonwload/index.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+            const fileUrl = await getFileUrl(fileId);
+            linkRef.setAttribute('href', fileUrl);
+            linkRef.click();
+        } catch (err) {
+            cogoToast.error(<div> לא ניתן להוריד את הקובץ </div>)
+            console.log(err);
+        }
+
+    }, files)
```
Comment (nachtiomer) [inline src/components/FilesDonwload/index.js:24]: files should be inside of array like [files]

**Pull request merged**